core/modules/wiki.js

```diff
@@ -700,21 +700,44 @@ exports.getTiddlerAsJson = function(title) {
 	}
 };
 
-exports.getTiddlersAsJson = function(filter,spaces) {
+/*
+Options:
+spaces: number of spaces
+escapeUnsafeScriptCharacters: true to force escaping of characters that cannot be embedded in HTML
+*/
+exports.getTiddlersAsJson = function(filter,options) {
+	options = options || {};
+	if(typeof options === "string") {
+		options = {spaces: options};
+	}
 	var tiddlers = this.filterTiddlers(filter),
-		spaces = (spaces === undefined) ? $tw.config.preferences.jsonSpaces : spaces,
-		data = [];
-	for(var t=0;t<tiddlers.length; t++) {
-		var tiddler = this.getTiddler(tiddlers[t]);
+		spaces = (options.spaces === undefined) ? $tw.config.preferences.jsonSpaces : options.spaces,
+		data = [],
+		t,fields,tiddler;
+	for(t=0;t<tiddlers.length; t++) {
+		tiddler = this.getTiddler(tiddlers[t]);
 		if(tiddler) {
-			var fields = new Object();
+			fields = new Object();
 			for(var field in tiddler.fields) {
 				fields[field] = tiddler.getFieldString(field);
 			}
 			data.push(fields);
 		}
 	}
-	return JSON.stringify(data,null,spaces);
+	var json = JSON.stringify(data,null,spaces);
+	if(options.escapeUnsafeScriptCharacters) {
+		function escapeUnsafeChars(unsafeChar) {
+			return {
+				"<":      "\\u003C",
+				">":      "\\u003E",
+				"/":      "\\u002F",
+				"\u2028": "\\u2028",
+				"\u2029": "\\u2029"
+			}[unsafeChar];
+		}
+		json = json.replace(/</g,escapeUnsafeChars);
```

Comment:
It looks like only the < char is replaced. Did you intend to replace the other chars listed in the escapeUnsafeChars function as well?

Maybe like this?

    json.repace(/<|>|\/|\\u2028|\\u2029/,escapeUnsafeChars);

Reply:
If not, then what's the point of listing them in the function?

Reply:
Hi @simonbaird apologies, I should have clarified: I was/am a bit uncertain whether just escaping `<` is sufficient, and so implemented multiple character replacement, and then ran cross browser tests. Once we confirm that `<` is sufficient, then I'll clean up the code.

Reply:
Ack, thanks.